README.md

```diff
@@ -1,12 +1,301 @@
+        asn: 64512
+        port: 180
+        bfdProfile: defaultprofile
+    bfdProfiles:
+      - name: defaultprofile
+```
+
+#### Advertising a prefix to a given neighbor
+
+By default, the prefixes configured as part of a router configuration are not advertised. In order to advertise them,
+the `toAdvertise` section must be filled.
+
+A subset of the prefixes can be advertised:
+
+```yaml
+    - address: 172.18.0.6
+    asn: 4200000000
+    port: 179
+    toAdvertise:
+        allowed:
+        prefixes:
+        - 192.168.2.0/24
+```
+
+or all the prefixes configured in the router can be advertised:
+
+```yaml
+    - address: 172.30.0.3
+    asn: 4200000000
+    ebgpMultiHop: true
+    port: 180
+    toAdvertise:
+        allowed:
+        mode: all
+```
+
+#### Receiving prefixes from a given neighbor
+
+By default, no prefixes advertised by a neighbour are processed.
+
+In order to process them, a `toReceive` section must be filled.
+
+A subset of the prefixes can be received:
+
+```yaml
+    - address: 172.18.0.5
+    asn: 64512
+    port: 179
+    toReceive:
+        allowed:
+        prefixes:
+        - 192.168.2.0/24
+        - 192.169.2.0/24
+```
+
+or FRR can be configured to handle all the prefixes announced:
+
+```yaml
+    - address: 172.18.0.5
+    asn: 64512
+    port: 179
+    toReceive:
+        allowed:
+        mode: all
+```
+
+### Adding a raw configuration
+
+In order to facilitate experimentation and to fill gaps quickly, it is possible to set a piece of raw
+FRR configuration that is appended to the rendered one. In order to do so, the `rawConfig` field
+must be set:
+
+```yaml
+apiVersion: frrk8s.metallb.io/v1beta1
+kind: FRRConfiguration
+metadata:
+  name: test1
+  namespace: frr-k8s-system
+spec:
+  raw:
+    priority: 5
+    rawConfig: |-
+      router bgp 64512
+        neighbor 172.18.0.5 remote-as 4200000000
+        neighbor 172.18.0.5 timers 0 0
+        neighbor 172.18.0.5 port 179
+```
+
+The priority field sets the order in case multiple configurations are merged together.
+
+### Node selector
+
+A node selector field drives the nodes where the given configuration is applied.
+
+```yaml
+apiVersion: frrk8s.metallb.io/v1beta1
+kind: FRRConfiguration
+metadata:
+  name: test
+  namespace: frr-k8s-system
+spec:
+  bgp:
+    routers:
+    - asn: 64512
+  nodeSelector:
+    labelSelector:
+      foo: "bar"
+```
+
+No node selector means the configuration is applied to all the nodes where the daemon is running:
+
+### How multiple configurations are merged together
+
+Multiple actors may add configurations selecting the same node. In this case, the configurations are merged together.
+
+The way `frr-k8s` configurability is intended is, each configuration can only extend others. This means that it is
+allowed (and possible) to add a new neighbor to a router, or to advertise an additional prefix to a neighbor, but there
+is no way to remove a component added by another configuration.
+
```

Comment:
I'd add something about each configuration being independent and cannot rely on others, and in case of overlap the more permissive one takes precedence (such as allowall in incoming routes)